optics_functions/coupling.py

```diff
@@ -0,0 +1,208 @@
+"""
+Coupling
+********
+
+Functions to calculate coupling from twiss dataframes.
+
+"""
+import logging
+from contextlib import suppress
+from typing import Sequence
+
+import numpy as np
+from tfs import TfsDataFrame
+
+from optics_functions.constants import (ALPHA, BETA, GAMMA,
+                                        X, Y, TUNE, DELTA,
+                                        MINIMUM, PI2, PHASE_ADV, S, LENGTH)
+from optics_functions.rdt import calculate_rdts
+from optics_functions.utils import split_complex_columns, timeit
+
+COUPLING_RDTS = ['F1001', 'F1010']
+
+
+LOG = logging.getLogger(__name__)
+
+
+def coupling_from_rdts(df: TfsDataFrame, real: bool = False, **kwargs):
```

Comment:
shouldn't there be also an append option to add RDT columns to the input tfs-df?

Reply:
No. It just adds input parameters, makes the code unnecessarily more complicated, more ifs. modifies the dataframe inplace (which I dont like) leaves open if it should really do that inplace or return a copy of the full input df and return `None`. etc.

All of which can be accomplished by just doing
`df[rdts] = calculate_rdts(df, rdts=rdts)`

Reply:
added example to Readme